### PR TITLE
HPCC-26123 React WU Details missing actions

### PR DESCRIPTION
**Pull request by jeclrsg**
Signed-off-by: Jeremy Clements <jeremy.clements@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: esp/src/src-react/components/WorkunitDetails.tsx [hunk 1]
```diff
@@ -84,6 +90,13 @@ export const WorkunitDetails: React.FunctionComponent<WorkunitDetailsProps> = ({
         description !== workunit.Description ||
         _protected !== workunit.Protected
     );
+    const canDelete = workunit && (
+        _protected !== workunit.Protected ||
+        999 !== workunit.StateID ||
+        workunit.Archived
+    );
+    const canDeschedule = workunit && workunit.EventSchedule === 2;
+    const canReschedule = workunit && workunit.EventSchedule === 1;
```
Comment (GordonSmith) [inline esp/src/src-react/components/WorkunitDetails.tsx:99]: `workunit?.EventSchedule === 1;`

File: esp/src/src-react/components/WorkunitDetails.tsx [hunk 2]
```diff
@@ -104,12 +123,88 @@ export const WorkunitDetails: React.FunctionComponent<WorkunitDetailsProps> = ({
             }
         },
         {
-            key: "copy", text: nlsHPCC.CopyWUID, iconProps: { iconName: "Copy" },
+            key: "delete", text: nlsHPCC.Delete, iconProps: { iconName: "Delete" }, disabled: !canDelete,
             onClick: () => {
-                navigator?.clipboard?.writeText(wuid);
+                if (confirm(nlsHPCC.YouAreAboutToDeleteThisWorkunit)) {
+                    workunit?.delete();
+                    pushUrl("/workunits/");
```
Comment (GordonSmith) [inline esp/src/src-react/components/WorkunitDetails.tsx:130]: For consistency:  `pushUrl("/workunits");`
`

File: esp/src/src-react/components/WorkunitDetails.tsx [hunk 3]
```diff
@@ -104,12 +123,88 @@ export const WorkunitDetails: React.FunctionComponent<WorkunitDetailsProps> = ({
...
+        {
+            key: "clone", text: nlsHPCC.Clone, disabled: workunit?.Archived || !workunit?.isComplete() || workunit?.isDeleted(),
+            onClick: () => {
+                // there seems to be an issue with clone() returning a 404 when
+                // attempting to refresh / query the status of the new workunit
+                // workunit?.clone().then(wu => {
+                //     if (wu && wu.Wuid) {
+                //         pushUrl(`/workunits/${wu?.Wuid}`);
+                //     }
+                // });
```
Comment (GordonSmith) [inline esp/src/src-react/components/WorkunitDetails.tsx:174]: Can you open an issue at `@hpcc-js/comms`?

File: esp/src/src-react/components/WorkunitDetails.tsx [hunk 4]
```diff
@@ -104,12 +123,88 @@ export const WorkunitDetails: React.FunctionComponent<WorkunitDetailsProps> = ({
...
+                const formData = new FormData();
+                formData.append("Wuids", wuid);
+                formData.append("ResetWorkflow", "0");
+                formData.append("CloneWorkunit", "1");
+                fetch("/WsWorkunits/WUResubmit.json", {
+                    method: "POST",
+                    body: formData
+                })
```
Comment (GordonSmith) [inline esp/src/src-react/components/WorkunitDetails.tsx:182]: If there isn't a resubmit method on workunit, you could add it into `@hpcc-js/comms`?

File: esp/src/src-react/components/forms/PublishQuery.tsx [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+                        }) => <Checkbox name={fieldName} checked={value} onChange={onChange} label={nlsHPCC.UpdateSuperFiles} /> }
+                    />
+                </div>
+            </Stack>
+            <Stack horizontal horizontalAlign="space-between" verticalAlign="end" styles={buttonStackStyles}>
+                <PrimaryButton text={nlsHPCC.Publish} onClick={handleSubmit(onSubmit)} />
+            </Stack>
+        </div>
+    </Modal>;
+};
```
Comment (GordonSmith) [inline esp/src/src-react/components/forms/PublishQuery.tsx:244]: The styles should be inside a "useMemo"

File: esp/src/src-react/components/forms/SlaveLogs.tsx [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+                const url = `/WsWorkunits/WUFile?Wuid=${wuid}&Type=ThorSlaveLog&Process=${data.ThorProcess}` +
+                    `&ClusterGroup=${clusterGroup}&LogDate=${thorLogDate}&SlaveNumber=${data.SlaveNumber}&Option=${data.FileFormat}`;
+                window.open(url);
+                setShowForm(false);
+            },
+            err => {
+                console.log(err);
+            }
+        )();
+    };
```
Comment (GordonSmith) [inline esp/src/src-react/components/forms/SlaveLogs.tsx:54]: Should be in a useCallback

File: esp/src/src-react/components/forms/SlaveLogs.tsx [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+
+    const titleId = useId("title");
+
+    const closeForm = () => {
+        setShowForm(false);
+        reset(defaultValues);
+    };
```
Comment (GordonSmith) [inline esp/src/src-react/components/forms/SlaveLogs.tsx:61]: useCallback

File: esp/src/src-react/components/forms/SlaveLogs.tsx [hunk 8]
```diff
@@ -0,0 +1,224 @@
...
+
+    const dragOptions: IDragOptions = {
+        moveMenuItemText: "Move",
+        closeMenuItemText: "Close",
+        menu: ContextualMenu,
+    };
```
Comment (GordonSmith) [inline esp/src/src-react/components/forms/SlaveLogs.tsx:67]: useMemo

File: esp/src/src-react/components/forms/SlaveLogs.tsx [hunk 9]
```diff
@@ -0,0 +1,224 @@
...
+
+    const theme = getTheme();
```
Comment (GordonSmith) [inline esp/src/src-react/components/forms/SlaveLogs.tsx:69]: useMemo

File: esp/src/src-react/components/forms/SlaveLogs.tsx [hunk 10]
```diff
@@ -0,0 +1,224 @@
...
+            padding: "4px",
+            border: `1px solid ${theme.palette.themeDark}`
+        },
+        twoColumnTable: {
+            marginTop: "14px",
+            "selectors": {
+                "tr": { marginTop: "10px" }
+            }
+        }
+    });
```
Comment (GordonSmith) [inline esp/src/src-react/components/forms/SlaveLogs.tsx:122]: Styles in a useMemo

**Review by GordonSmith — CHANGES_REQUESTED** [review]

File: esp/src/src-react/components/WorkunitDetails.tsx (hunk 3 above)
Comment (jeclrsg) [reply esp/src/src-react/components/WorkunitDetails.tsx:173]: Opened https://github.com/hpcc-systems/Visualization/issues/3829

File: esp/src/src-react/components/WorkunitDetails.tsx (hunk 4 above)
Comment (jeclrsg) [reply esp/src/src-react/components/WorkunitDetails.tsx:181]: There is one already, but it doesn't accept a param for CloneWorkunit. This was just a workaround for the issue with clone()

File: esp/src/src/nls/hpcc.ts [hunk 11]
```diff
@@ -916,6 +920,10 @@ export = {
         ValidatePackageMap: "Validate Package Map",
         ValidateResult: "=====Validate Result=====\n\n",
         ValidateResultHere: "(Validation result)",
+        ValidationErrorEnterNumber: `Enter a valid number`,
+        ValidationErrorNumberGreater: `Cannot be greater than`,
+        ValidationErrorNumberLess: `Cannot be less than`,
+        ValidationErrorRequired: `This field is required`,
```
Comment (GordonSmith) [inline esp/src/src/nls/hpcc.ts:926]: Wrong quotes?

**Review by GordonSmith — CHANGES_REQUESTED** [review]

File: esp/src/src-react/components/WorkunitDetails.tsx (hunk 4 above)
Comment (GordonSmith) [reply esp/src/src-react/components/WorkunitDetails.tsx:181]: @jeclrsg - This fix should be released now.

**Review by GordonSmith — APPROVED** [review]

**Pull request merged**